pandas/core/groupby.py

```diff
@@ -1472,7 +1479,7 @@ def pad(self, limit=None):
         Series.fillna
         DataFrame.fillna
         """
-        return self.apply(lambda x: x.ffill(limit=limit))
+        return self.apply('ffill', limit=limit)
```

Comment:
Rather than calling `apply`, pattern so far has been to define the method directly `Groupby.cumsum`, etc

Reply:
Tied to the comment below, I added `_cython_apply` as its own function to handle dispatching back to the Cython layer, which this `apply` tries first before falling back to the current method of iterating over the groups and applying the method directly from there. Items like `Groupby.cumsum` call `_cython_transform` in a similar fashion, which I was trying to emulate here but I was hesitant to send these methods down the transform route because of the complexity that would cause there